src/merge.ts

```diff
@@ -0,0 +1,135 @@
+import type { Config, Octokit, File, FrontMatter } from './types';
+import type { Repository } from '@octokit/webhooks-types';
+import localConfig from './localConfig';
+import fm from 'front-matter';
+import yaml from 'js-yaml';
+
+async function generateEIPNumber(octokit: Octokit, repository: Repository) {
+    // Get all EIPs
+    const eips = (await octokit.rest.repos.getContent({
+        owner: repository.owner.login,
+        repo: repository.name,
+        path: 'EIPS'
+    })).data as any[];
+
+    // Get all EIP numbers
+    const eipNumbers = eips
+        .filter(eip => eip.name.startsWith('eip-'))
+        .map(eip => Number(eip.name.split('-')[1]));
+
+    // Find the biggest EIP number
+    const eipNumber = Math.max(...eipNumbers);
+
+    // Add a random number from 1-5 to the EIP number
+    // This is to prevent conflicts when multiple PRs are merged at the same time, and to prevent number gaming
+    return eipNumber + Math.floor(Math.random() * 5) + 1;
+}
+
+export async function performMergeAction(octokit: Octokit, _: Config, repository: Repository, pull_number: number, files: File[]) {
+    // Fetch PR data
+    let pull_request = (await octokit.rest.pulls.get({
+        owner: repository.owner.login,
+        repo: repository.name,
+        pull_number: pull_number
+    })).data;
+    const title = pull_request.title;
+
+    // Modify EIP data when needed
+    let newFiles = [];
+    for (let file of files) {
+        file = { ...file };
+        if (file.filename.endsWith('.md')) {
+            // Parse file
+            const fileContent = file.contents as string;
+            const fileData = fm(fileContent);
+            const frontmatter = fileData.attributes as FrontMatter;
+
+            // Check if EIP number needs setting
+            if (!frontmatter.eip) {
+                let eip = await generateEIPNumber(octokit, repository);
+
+                frontmatter.eip = `${eip}`;
+                file.filename = `EIPS/eip-${eip}.md`;
+            }
```

Comment:
Newly created EIPs with an `eip` header should require mandatory merging.

Reply:
Wdym?

Reply:
edited, sorry.

Reply:
> Newly created EIPs with an eip header should require manual merging.

Are you sure you want that?

Reply:
Going to assume you meant editor approval, since we want to deprecate manual merging.

Reply:
No, not approval. Manual merging. It's too easy for an editor to miss that an EIP has a manually assigned number.

Reply:
Actually, this would fit better into eipw.